### PR TITLE
add ToC to docs

### PR DESCRIPTION
**Pull request by ilaflamme-va**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,13 @@
 # health-apis-deployer
+  + [Goals](#goals)
```
Comment (jhulbert-va) [inline README.md]: Can we add a header here that says `Contents` or something like that? That way, theres no question as to what this is.

File: deployment-unit.md [hunk 2]
```diff
@@ -1,5 +1,17 @@
 # Deployment Unit
 
++ [Product Repository Structure](#product-repository-structure)
```
Comment (jhulbert-va) [inline deployment-unit.md]: Same.

**Comment by mramirez-va** [comment]
`Defining Resource Requests/Limits` and `Blue/Green Deployment Process` links don't work for me 😢

**Comment by jhulbert-va** [comment]
> `Defining Resource Requests/Limits` and `Blue/Green Deployment Process` links don't work for me

Try it again plz.

**Pull request merged**